src/sources/internal.rs

```diff
@@ -0,0 +1,178 @@
+use crate::{
+    event::metric::{Metric, MetricKind, MetricValue},
+    topology::config::{DataType, GlobalOptions, SourceConfig, SourceDescription},
+    Event,
+};
+use chrono::Utc;
+use futures::{
+    compat::Future01CompatExt,
+    future::{FutureExt, TryFutureExt},
+};
+use futures01::{sync::mpsc, Future, Sink, Stream};
+use metrics_core::Key;
+use metrics_runtime::{Controller, Measurement};
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use stream_cancel::Tripwire;
+use tokio::timer::Interval;
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+pub struct InternalConfig;
+
+inventory::submit! {
+    SourceDescription::new::<InternalConfig>("internal")
+}
+
+#[typetag::serde(name = "internal")]
```

Comment:
Not the biggest fan of this name but I guess it works. I also don't think we need to type tag it either? This should prob get its config from the global settings and be added by hand to the topology.

Reply:
Yeah, I mentioned giving this a real name as something we need to do before merging.

I'm a little less clear on how the config should work. I can see using global config since this is logically a "singleton" source (though it technically doesn't need to be). But that would involve giving it a special name to be referred to in other pipelines? We'll want to think about the UX here, especially default behaviors.